libfreerdp/core/gateway/rdg.c

```diff
@@ -323,34 +359,313 @@ static BOOL rdg_write_packet(rdpRdg* rdg, wStream* sPacket)
 	return TRUE;
 }
 
+static BOOL rdg_write_websocket(rdpRdg* rdg, wStream* sPacket)
+{
+	size_t s;
+	int status;
+	wStream* sWS;
+
+	uint32_t maskingKey;
+
+	int streamPos;
+
+	s = Stream_Length(sPacket);
+	Stream_SetPosition(sPacket, 0);
+
+	if (s > INT_MAX)
+		return FALSE;
+
+	if (s < 126)
+		sWS = Stream_New(NULL, s + 6); // 2 byte "mini header" + 4 byte masking key
+	else if (s < 0x10000)
+		sWS = Stream_New(NULL, s + 8); // 2 byte "mini header" + 2 byte length + 4 byte masking key
+	else
+		sWS = Stream_New(NULL, s + 14); // 2 byte "mini header" + 8 byte length + 4 byte masking key
+
```

Comment:
Just nitpicking, but perhaps the line above could just compute an allocation size and the effective `Stream_New` would be done here.